### PR TITLE
Dead No Gurney changes.

### LINKED ISSUE #529: Dead No Gurney Message
From #519 
Will need to change how health is modified so that it can trigger events.
From Discord - keep in mind possibility for scripts to register to accessor functions in the future.

### PR DESCRIPTION
**Pull request by Magentah**
Closes #529  .

Additions/modifications proposed in this pull request:
- Health changes are because I initially thought it would be triggered by health reaching 0, but it's triggered by a client event. I can undo them if needed.
- Added client event AwaitingDeadNoGurney and server event DeadNoGurney
- Added test command, which just calls the MapInstance event handler.

### CODE REVIEW

**Comment by Magentah** [comment]
I've never actually used C++ before outside of small console apps, so there's probably some issues with what I wrote..

File: Projects/CoX/Servers/MapServer/DataHelpers.cpp [hunk 1]
```diff
@@ -526,16 +526,16 @@ void setCombatLevel(Character &c, uint32_t val)
...
-}
-
 void setEnd(Character &c, float val)
 {
     c.m_char_data.m_current_attribs.m_Endurance = std::max(0.0f, std::min(val,c.m_max_attribs.m_Endurance));
 }
 
+void modifyHealth(Entity *tgt, float amount, bool send_events)
+{    
+    tgt->m_char->setHealth(amount);
```
Comment (dracc) [inline Projects/CoX/Servers/MapServer/DataHelpers.cpp]: Afaict, you're not modifying health as much as you're setting it to a value. But you use this function (`modifyHealth`) as if it did `currentHealth += amount`.

Comment (Magentah) [reply Projects/CoX/Servers/MapServer/DataHelpers.cpp]: Ah good catch, missed that.

File: Projects/CoX/Servers/MapServer/Events/DeadNoGurney.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+#pragma once
+#include "GameCommandList.h"
+
+#include "MapEvents.h"
+#include "MapLink.h"
+
+#include <QtCore/QString>
+
+class DeadNoGurney : public GameCommand
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/Events/DeadNoGurney.h]: Can you please add unit test comments to this class and it's members. You can see an example in PowerSystemEvents.h

`// [[ev_def:type]]` above the class, and
`// [[ev_def:field]]` above each member variable (you have none)

File: Projects/CoX/Servers/MapServer/MapEvents.h [hunk 3]
```diff
@@ -259,6 +259,20 @@ class EnterDoor final : public MapLinkEvent
     }
 };
 
+class AwaitingDeadNoGurney final : public MapLinkEvent
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/MapEvents.h]: Same here please `// [[ev_def:type]]`

**Comment by Magentah** [comment]
Added the comments.

File: Projects/CoX/Servers/MapServer/MapEvents.h (hunk 3 above)
Comment (broxen) [reply Projects/CoX/Servers/MapServer/MapEvents.h]: Oops. Can we add this ev_def comment here as well? Thanks!

File: Projects/CoX/Servers/MapServer/MapInstance.cpp [hunk 4]
```diff
@@ -2204,6 +2207,17 @@ void MapInstance::on_recv_new_power(RecvNewPower *ev)
     addPower(session.m_ent->m_char->m_char_data, ev->ppool);
 }
 
+void MapInstance::on_awaiting_dead_no_gurney(AwaitingDeadNoGurney *ev)
+{
+    MapClientSession &session(m_session_store.session_from_event(ev));
+    session.m_ent->m_client->addCommandToSendNextUpdate(std::unique_ptr<DeadNoGurney>(new DeadNoGurney()));
+}
+
+void MapInstance::on_awaiting_dead_no_gurney_test(AwaitingDeadNoGurney *ev, MapClientSession &session)
```
Comment (broxen) [inline Projects/CoX/Servers/MapServer/MapInstance.cpp]: I'm wondering if we should move this method to DataHelpers the way we have `sendFloatingInfo()`

Then, in a possible Lua future we can expose that to the scripts without needing MapInstances and MapClientSession

Comment (Magentah) [reply Projects/CoX/Servers/MapServer/MapInstance.cpp]: `evAwaitingDeadGurney` is a client -> server event, would that still work? `evFloatingInfo` is server -> client, the same as `evDeadNoGurney`, which is already in DataHelpers.

**Pull request merged**